### PR TITLE
*: refactor clustermembercontroller

### PR DESCRIPTION
**Pull request by alaypatel07**
This sync loop is a giant monolith, first attempt in refactoring this.
Probably needs to break into two controllers in the future, but trying to
get there organically.

### CODE REVIEW

**Comment by alaypatel07** [comment]
wip because I need to review adding events

File: pkg/operator/clustermembercontroller/clustermembercontroller.go [hunk 1]
```diff
@@ -38,6 +37,8 @@ const (
 	EtcdEndpointName               = "etcd"
 	ConditionBootstrapSafeToRemove = "BootstrapSafeToRemove"
 	ConditionBootstrapRemoved      = "BootstrapRemoved"
+	waitForKubeContainerNumber     = 0
+	EtcdClusterSize                = 3
```
Comment (hexfusion) [inline pkg/operator/clustermembercontroller/clustermembercontroller.go]: This feels like it should be based on the expected number of master nodes as reported by machine api or similar? I understand for testing this works but a use-case for Verizon is installing a single master. Then a short time afterward reprovision the bootstrap node as a master and scale up to 3. Just something we should consider.

Comment (alaypatel07) [reply pkg/operator/clustermembercontroller/clustermembercontroller.go]: yes, I planned to have a todo, that number needs to come from somewhere, maybe the installer. For now, is it okay if we make it equal to the number of masters (querying from kube-api and not hardcoded as it currently is)?

**Comment by alaypatel07** [comment]
closing because big pictures remove this controller

**Pull request closed without merging**